google/cloud/storage/_signing.py

```diff
@@ -666,3 +665,33 @@ def _sign_message(message, access_token, service_account_email):
 
     data = json.loads(response.data.decode("utf-8"))
     return data["signature"]
+
+
+def _url_encode(query_params):
+    """Encode query params into URL.
+
+    :type query_params: dict
+    :param query_params: Query params to be encoded.
+
+    :rtype: str
+    :returns: URL encoded query params.
+    """
+    params = []
+    for name, value in query_params.items():
+        params.append(_quote_param(name) + "=" + _quote_param(value))
+
+    return "&".join(sorted(params))
```

Comment:
First encode, then sort, then join into single one `str`

Reply:
@frankyn is sorting needed here? The old code did the same though I wouldn't think the order was important?

Reply:
@crwilcox, @frankyn, yeah, that's mentioned in the docs. It's why I moved sorting from the start of the encoding code to the end - to strictly follow these lines

![image](https://user-images.githubusercontent.com/26030071/74226549-cc6e2a80-4ccd-11ea-8fc3-9ba195e62c03.png)